File: bin/roll_dir/no_comments/roll.cxx
Int_t roll( Int_t k, Int_t n, Int_t b = 0 ) {
Int_t total(0);
TVectorF rolls(k);
Int_t j;
for ( j=0; j<k; j++ ) {
  rolls[j] = 1 + gRandom->Integer(n-1);
}
total = rolls.Sum() + b;
printf("%dd%d + {%d}  ==>  ", k, n, b);
for ( j=0; j<k; j++ ) {
  printf("%d ", rolls[j]);
  if ( j < k-1 ) {
    printf("+ ");
  } else {
    printf("+ {%d}  =  %d\n", b, total);
  }
}
return total;
}
Int_t roll( Int_t n ) {
  Int_t total = 1 + gRandom->Integer(n-1);
  printf("%d\n", total);
  return total;
}

File: bin/roll_dir/no_comments/roll.m
function [ ] = roll( n, k, b )
if nargin < 3
  b = 0;
end
if nargin < 2
  k = 1;
end
rolls = zeros(1,k);
tot = 0;
for i = [1:k]
  rolls(1,i) = ceil( n * rand );
end %for
tot = sum(rolls) + b;
printf('\nRolls:\n')
disp(rolls)
if b ~= 0
  printf('Bonus:\t%i\n\n', b)
end
printf('TOTAL:\t%i\n', tot)
printf('\n\n')
end
